sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1266,6 +1273,25 @@ static bool compatibleWithDevice(RTDeviceBinaryImage *BinImage,
   return (0 == SuitableImageID);
 }
 
+kernel_id ProgramManager::getSYCLKernelID(const std::string &KernelName) {
+  std::lock_guard<std::mutex> Guard(Sync::getGlobalLock());
```

Comment:
Is it possible to employ a less global lock to reduce serialization of different operations?

Reply:
Absolutely. I can introduce a new lock specific to accessing the kernel ID cache. It would need to be acquired inside a block using the global lock, so we would have to be careful not to introduce code that acquires the global lock while holding the new lock. I'll make sure to make a note of it.

Reply:
I've added `m_KernelIDsMutex` for this purpose. It has a note to enforce a locking hierarchy.